src/modules/Nsswitch.rb

```diff
@@ -27,78 +27,95 @@
 #
 # $Id$
 require "yast"
+require "cfa/nsswitch"
 
 module Yast
   class NsswitchClass < Module
     def main
-
       Yast.import "Message"
       Yast.import "Report"
     end
 
     # Reads a database entry from nsswitch_conf and returns it as a list
-    # @param [String] db eg. "passwd"
-    # @return   eg. ["files", "nis"]
-    def ReadDb(db)
-      db_s = Convert.to_string(
-        SCR.Read(Builtins.add(path(".etc.nsswitch_conf"), db))
-      )
-      db_s = "" if db_s == nil
-      Builtins.filter(Builtins.splitstring(db_s, " \t")) { |s| s != "" }
+    #
+    # @see CFA::Nsswitch#services_for
+    #
+    # @param db_name [String] database entry name, e.g. "passwd"
+    def ReadDb(db_name)
+      cfa_model.services_for(db_name)
     end
 
 
     # Writes a database entry as a list to nsswitch_conf
-    # @param [String] db eg. "passwd"
-    # @param [Array<String>] entries eg. ["files", "nis"]
-    # @return success?
-    def WriteDb(db, entries)
-      entries = deep_copy(entries)
-      # if there are no entries, delete the key using nil
-      entry = Builtins.mergestring(entries, " ")
-      SCR.Write(
-        Builtins.add(path(".etc.nsswitch_conf"), db),
-        entry == "" ? nil : entry
-      )
+    #
+    # @see CFA::Nsswitch#update_entry
+    #
+    # @param db_name [String] database entry name, e.g. "passwd"
+    # @param services [Array<String>] service specs, e.g. ["files", "nis"]
+    def WriteDb(db_name, services)
+      cfa_model.update_entry(db_name, services)
     end
-    # Configures the name service switch for autofs
-    # according to chosen settings
-    # @param [Boolean] start autofs and service (ldap/nis) should be started
-    # @param [String] source for automounter data (ldap/nis)
-    # @return success?
-    def WriteAutofs(start, source)
-      ok = true
 
+    # Configures the name service switch for autofs according to chosen settings
+    #
+    # @see #Write
+    #
+    # @param start [Boolean] whether autofs and service (ldap/nis) should be started
+    # @param source [String] source for automounter data (ldap/nis)
+    #
+    # @return [Boolean] true on success; false otherwise
+    def WriteAutofs(start, source)
       # nsswitch automount:
-      # bracket options not allowed, order does not matter
-      automount_l = ReadDb("automount")
-      enabled = Builtins.contains(automount_l, source)
+      # bracket options not allowed
+      automount_services = cfa_model.services_for("automount")
+      enabled = automount_services.include?(source)
+
+      # enable it if it is not enabled yet and both services run
```

Comment:
This code is too complex for such a simple thing (unless I am missing something).

```ruby
if start
  amount_services |= [source]
else
  amount_services -= [source]
end
```

That's it: add the service if it should be started or remove if not. I guess that the `enabled` variable it is used basically to do not add the service twice (or not try to remove if it is not there). But I would say that you can get rid of it.

Reply:
Thanks so much for helping here. What you say makes sense, but as I didn't fully understand the original piece of code I decided to go ahead in a conservative way.

BTW, thanks to your comment I realize that there is a leftover of my own attempts to reduce/simplify the original code. 

So, thanks twice :)